pkg/deploy/controller/deployment/controller.go

```diff
@@ -144,13 +144,18 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		// Generate a deployer pod spec.
 		deployerPod, err := c.makeDeployerPod(deployment)
 		if err != nil {
-			return fatalError(fmt.Sprintf("couldn't make deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err))
+			nextStatus = deployapi.DeploymentStatusFailed
+			updatedAnnotations[deployapi.DeploymentStatusReasonAnnotation] = err.Error()
+			c.emitDeploymentEvent(deployment, kapi.EventTypeWarning, "FailedDecoding", err.Error())
```

Comment:
The only error returned from `makeDeployerPod` is a failure to decode the annotation that contains the encoded deployment config. Not sure anyone has hit it ever but it's possible. Currently, we don't provide any workaround to it but potentially we could.